docs/how-to/hip_runtime_api/multi_device.rst

```diff
@@ -0,0 +1,287 @@
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Attempt to use deviceData0 on device 1 (This will not work as deviceData0 is allocated on device 0)
+        HIP_CHECK(hipSetDevice(deviceId1));
+        hipError_t err = hipMemcpy(deviceData1, deviceData0, size, hipMemcpyDeviceToDevice); // This should fail
+        if (err != hipSuccess)
+        {
+            std::cout << "Error: Cannot access deviceData0 from device 1, deviceData0 is on device 0" << std::endl;
+        }
+
+        // Copy result from device 0
+        double hostData0[1024];
+        HIP_CHECK(hipSetDevice(deviceId0));
+        HIP_CHECK(hipMemcpy(hostData0, deviceData0, size, hipMemcpyDeviceToHost));
+
+        // Copy result from device 1
+        double hostData1[1024];
+        HIP_CHECK(hipSetDevice(deviceId1));
+        HIP_CHECK(hipMemcpy(hostData1, deviceData1, size, hipMemcpyDeviceToHost));
+
+        // Display results from both devices
+        std::cout << "Device 0 data: " << hostData0[0] << std::endl;
+        std::cout << "Device 1 data: " << hostData1[0] << std::endl;
+
+        // Free device memory
+        HIP_CHECK(hipFree(deviceData0));
+        HIP_CHECK(hipFree(deviceData1));
+
+        return 0;
+    }
+
+
+Stream and event behavior
+===============================================================================
+
+In a multi-device system, streams and events are essential for efficient
+parallel computation and synchronization. Streams enable asynchronous task
+execution, allowing multiple devices to process data concurrently without
+blocking one another. Events provide a mechanism for synchronizing operations
+between devices, ensuring that tasks on one device are completed before the
+next device starts its work. This coordination prevents race conditions and
+optimizes data flow in multi-GPU systems. Together, streams and events maximize
+performance by enabling parallel execution, load balancing, and effective
+resource utilization across heterogeneous hardware.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    __global__ void simpleKernel(double *data)
+    {
+        int idx = blockIdx.x * blockDim.x + threadIdx.x;
+        data[idx] = idx * 2.0;
+    }
+
+    int main()
+    {
+        double* deviceData;
+        size_t  size = 1024 * sizeof(*deviceData);
+
+        // Create a stream and events
+        hipStream_t stream;
+        hipEvent_t startEvent, stopEvent;
+
+        hipStreamCreate(&stream);
+        hipEventCreate(&startEvent);
+        hipEventCreate(&stopEvent);
+
+        // Allocate memory on device
+        hipMalloc(&deviceData, size);
+
+        // Record the start event
+        hipEventRecord(startEvent, stream);
+
+        // Launch the kernel asynchronously
+        simpleKernel<<<1000, 128, 0, stream>>>(deviceData);
+
+        // Record the stop event
+        hipEventRecord(stopEvent, stream);
```

Comment:
Is this example showing what the paragraph above describes? It seems like it is simply a kernel running on a device, and some wait state before capturing performance data?

Reply:
Thanks, you are right! I updated the code.